src/settings/components/ExternalDataSourcesConfig/ExternalDataSourcesSettings/ExternalDataSourcesSettings.js

```diff
@@ -0,0 +1,87 @@
+import { useState } from 'react';
+import PropTypes from 'prop-types';
+import { FormattedMessage } from 'react-intl';
+import arrayMutators from 'final-form-arrays';
+
+import ExternalDataSourcesView from '../ExternalDataSourcesView/ExternalDataSourcesView';
+import ExternalDataSourcesLookup from '../ExternalDataSourcesLookup/ExternalDataSourcesLookup';
+import ExternalDataSourcesFormModal from '../ExternaldataSourcesFormModal/ExternalDataSourcesFormModal';
+import ExternalDataSourceForm from '../ExternalDataSourceForm/ExternalDataSourceForm';
+
+const CREATING = 'create';
+const VIEWING = 'view';
+
+const ExternalDataSourcesSettings = ({
+  onDelete,
+  onCancel,
+  onSave,
+  externalKbs,
+  onSubmit,
+  initialValues
+}) => {
+  const [mode, setMode] = useState(VIEWING);
+  const [externalDataSource, setExternalDataSource] = useState();
+
+  return (
+    <>
+      <ExternalDataSourcesLookup
+        externalKbs={externalKbs}
+        onCreateClick={() => setMode(CREATING)}
+        onSelectedExternalDataSource={(_e, eds) => {
+          setExternalDataSource(eds);
+        }}
+      />
+      {
+        externalDataSource &&
+          <ExternalDataSourcesView
+            externalDataSourceId={externalDataSource.id}
+            externalKbs={externalKbs}
+            initialValues={initialValues}
+            onClose={() => setExternalDataSource()}
+            onDelete={onDelete}
+            onEditCancel={onCancel}
+            onSave={onSave}
+            onSubmit={onSubmit}
+          />
+      }
+      <ExternalDataSourcesFormModal
+        initialValues={mode === CREATING ?
```

Comment:
Because we've split them out, we now only need the initial values corresponding to "CREATE", so you can remove this if/else ternary and just use the create values.